### PR TITLE
Remove queries from knex error messages

### PR DESCRIPTION
**Pull request by owais**

### CODE REVIEW

File: src/plugins/hapi.js [hunk 1]
```diff
@@ -56,93 +57,64 @@ function createWrapDispatch (tracer, config) {
...
+  patchRequestGenerate('hapi', ['>=17.1 <19.0']),
+  patchGenerate('hapi', ['8.5 - 17.0'], 'lib/request.js', 'request'),
+  patchExecute('hapi', ['2 - 8.4'], 'lib/request.js'),
+  patchDispatch('hapi', ['7.2 - 16'], 'lib/connection.js'),
+  patchDispatch('@hapi/hapi', ['>=17.9 <19.0'], 'lib/core.js'),
+  patchDispatch('hapi', ['>=17 <19.0'], 'lib/core.js'),
+  patchDispatch('hapi', ['2 - 7.1'], 'lib/server.js')
 ]
+
+if (semver.gte(process.version, '12.0.0')) {
```
Comment (owais) [inline src/plugins/hapi.js:111]: Hapi `19.0` adds deps that only work on node>=12 so activating those patches breaks CI jobs that use older node version.

File: src/plugins/knex.js [hunk 2]
```diff
@@ -49,41 +55,37 @@ function createWrapRunnerQuery (tracer, config, usePromise) {
...
-                span.finish()
-              })
-          })
-        } else {
-          try {
-            return original.apply(this, arguments)
-          } catch (e) {
-            throw addError(span, e)
-          } finally {
-            span.finish()
```
Comment (owais) [inline src/plugins/knex.js:74]: We added promise based handling for newer version of knex some time ago and kept try..catch based handling around for older versions. Turns out promise based handling actually works with all versions.

File: lib/version.js [hunk 3]
```diff
@@ -1 +1 @@
-module.exports = '0.3.0'
+module.exports = '0.6.4'
```
Comment (rmfitzpatrick) [inline lib/version.js:1]: 😅

File: src/plugins/hapi.js [hunk 4]
```diff
@@ -56,93 +57,62 @@ function createWrapDispatch (tracer, config) {
   }
 }
 
-module.exports = [
-  {
-    name: '@hapi/hapi',
-    versions: ['>=17.9'],
+function patchRequestGenerate (name, versions) {
```
Comment (rmfitzpatrick) [inline src/plugins/hapi.js:60]: I know this boilerplate is inherited but the intermediate helpers don't provide much imo:

```
function patch (name, versions, file, property, method, prototypical) {
  return {
    name: name,
    versions: versions,
    file: file,
    patch (Obj, tracer, config) {
	    const target = prototypical ? Obj.prototype : Obj
      this.wrap(target, property, method(tracer, config))
    },
    unpatch (Obj) {
	    const target = prototypical ? Obj.prototype : Obj
      this.unwrap(target, property)
    }
  }
}

let patches = [
  patch('@hapi/hapi', ['>=17.9 <19.0'], 'lib/request.js', 'generate', createWrapGenerate, true),
  patch('hapi', ['>=17.1'], 'lib/request.js', 'generate', createWrapGenerate, true),
  patch('hapi', ['8.5 - 17.0'], 'lib/request.js', 'request', createWrapGenerate),
  patch('hapi', ['2 - 8.4'], 'lib/request.js', '_execute', createWrapExecute),
  patch('hapi', ['7.2 - 16'], 'lib/connection.js', '_dispatch', createWrapDispatch),
  patch('@hapi/hapi', ['>=17.9 <19.0'], 'lib/core.js', '_dispatch', createWrapDispatch),
  patch('hapi', ['>=17'], 'lib/core.js', '_dispatch', createWrapDispatch),
  patch('hapi', ['2 - 7.1'], 'lib/server.js', '_dispatch', createWrapDispatch)
]

if (semver.gte(process.version, '12.0.0')) {
  patches = patches.concat([
    patch('@hapi/hapi', ['>=19.0'], 'lib/request.js', 'generate', createWrapGenerate, true),
    patch('@hapi/hapi', ['>=19.0'], 'lib/core.js', '_dispatch', createWrapDispatch)
  ])
}
```

Comment (owais) [reply src/plugins/hapi.js:60]: I thought they made the array with `patch()` calls a bit more readable but I guess that goes into personal preference territory. Don't have any strong feelings about it. Can get rid of the intermediate ones.

Comment (owais) [reply src/plugins/hapi.js]: Updated.

**Comment by owais** [comment]
@rmfitzpatrick Updated. PTAL.

**Pull request merged**